analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotTestThisWithIsOperator.CSharp9.cs

```diff
@@ -21,7 +41,24 @@ public Program(object o)
                     break;
             }
 
-            var result = this switch // FN
+            switch (o)
+            {
+                case IDisposable:
+                    break;
+                case IEnumerable:
+                    break;
+                default:
+                    break;
+            }
+
+            var result = (this) switch // Noncompliant
```

Comment:
`(this)` is here to test the parenthesis removal? If yes, let's be explicit about it, because it looks like a copy-paste issue from switch statement on first look and inconsistent with the next `o switch` case.
```suggestion
            var result = (((this))) switch // Noncompliant
```